src/main/scala/com.snowplowanalytics.sauna/responders/urbanairship/UAResponder.scala

```diff
@@ -0,0 +1,103 @@
+package com.snowplowanalytics.sauna
+package responders
+package urbanairship
+
+import scala.io.Source.fromInputStream
+
+import java.io.InputStream
+
+import apis.urbanAirship
+
+import akka.actor.{ ActorRef, Props }
+
+import responders.Responder.FileAppeared
+
+  /**
+ * Does stuff for UrbanAirship Static List upload feature.
+ *
+ * @see https://github.com/snowplow/sauna/wiki/Urban-Airship-Responder-user-guide
+ * @param urbanairship Instance of UrbanAirship.
+ * @param logger A logger actor.
+ */
+
+class UAResponder(urbanairship: urbanAirship)(implicit logger: ActorRef) extends Responder {
+
+  import UAResponder._
+
+  val pathPattern =
+    """.*com\.urbanairship/
+      |static_lists/
+      |v1/
+      |tsv:\*/
+      |.+$
+    """.stripMargin
+      .replaceAll("[\n ]", "")
+
+  def process(fileAppeared: FileAppeared): Unit = {
+    import fileAppeared._
+    convertTSV(is)
+  }
+  
+/**
+   * Converts the TSV data and groups it in a map so as to make batched upload requests to UrbanAirship
+   *
+   * @param is the InputStream of the TSV file 
+   * @return a map of the data grouped by application keys and then by listNames
+   */
+
+  def convertTSV(is: InputStream): Unit = {
```

Comment:
Hello @rmanojcit!

I see a lot of improvements in your code, which means we can start our second refactoring tour.
Let's concentrate or attention now around this `convertTSV` function.  We have few problems with it.